staging/src/k8s.io/kubectl/pkg/cmd/attach/attach.go

```diff
@@ -128,7 +130,7 @@ type RemoteAttach interface {
 // DefaultAttachFunc is the default AttachFunc used
 func DefaultAttachFunc(o *AttachOptions, containerToAttach *corev1.Container, raw bool, sizeQueue remotecommand.TerminalSizeQueue) func() error {
 	return func() error {
-		restClient, err := restclient.RESTClientFor(o.Config)
+		restClient, err := restclient.RESTClientForConfigAndClient(o.Config, o.HttpClient)
```

Comment:
don't you have the same effect just moving the constructor before the closure?

```go
// DefaultAttachFunc is the default AttachFunc used
func DefaultAttachFunc(o *AttachOptions, containerToAttach *corev1.Container, raw bool, sizeQueue remotecommand.TerminalSizeQueue) func() error {
restClient, err := restclient.RESTClientFor(o.Config)
		if err != nil {
			return err
		}
	return func() error {
				req := restClient.Post().
			Resource("pods").
			Name(o.Pod.Name).
			Namespace(o.Pod.Namespace).
			SubResource("attach")
```

Reply:
Unfortunately not, since the config and client is not yet initialized